File: src/main/java/com/cuupa/classificator/services/kb/semantic/token/Not.java
package com.cuupa.classificator.services.kb.semantic.token;

import com.cuupa.classificator.services.kb.semantic.text.TextSearch;
import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;

public class Not extends Token {

	@Override
	public boolean match(String text) {
        TextSearch plaintext = new TextSearch(text);
		for (String token : tokenValue) {
			if (plaintext.contains(token)) {
				return false;
			}
		}
		return true;
	}

	@Override
	public int getDistance() {
		return 0;
	}

	@NotNull
	@Override
	protected Token clone() {
		Token token = new Not();
		token.setTokenValue(new ArrayList<>(tokenValue));
		return token;
	}

	@NotNull
	@Override
	public String toString() {
		return "NOT " + super.toString();
	}
}

File: src/main/java/com/cuupa/classificator/services/kb/semantic/token/Not.kt
package com.cuupa.classificator.services.kb.semantic.token

import com.cuupa.classificator.services.kb.semantic.text.TextSearch
import java.util.*

class Not : Token() {

    override fun match(text: String?): Boolean {
        val plaintext = TextSearch(text)
        for (token in tokenValue) {
            if (plaintext.contains(token)) {
                return false
            }
        }
        return true
    }

    override val distance: Int
        get() = 0

    override fun clone(): Token {
        val token: Token = Not()
        token.tokenValue = ArrayList(tokenValue)
        return token
    }

    override fun toString(): String {
        return "NOT " + super.toString()
    }

    companion object {
        const val name = "not"
    }
}